# XML - Lab

## Introduction

In this lab, you'll practice navigating XML data structures.

## Objectives
You will be able to:
* Use the XML module to load and parse XML data


## XML

In [2]:
import xml.etree.ElementTree as ET

### Create an XML tree and retrieve the root tag.

In [3]:
#Your code here
tree = ET.parse('nyc_2001_campaign_finance.xml')
root = tree.getroot()

### How many direct descendents does the root tag have?

In [8]:
#Your code here
len(root)

1

### How many different types of tags are there within the entire XML file?

In [18]:
# Your code here

#Complexity added to create dataframe in next step.

tag_count = {}
for element in root.iter():
    n = tag_count.get(element.tag, [0])
    tag_count[element.tag] = [n[0]+1]
print(tag_count)
len(tag_count.keys())

{'response': [1], 'row': [286], 'candid': [285], 'candname': [285], 'officeboro': [21], 'canclass': [285], 'election': [284], 'officecd': [284], 'officedist': [245], 'primarypay': [284], 'generalpay': [284], 'runoffpay': [284], 'totalpay': [284]}


13

### Create a DataFrame listing the number of each type of tag. 
Sort the DataFrame in descending order by the tag count. The first entry should demonstrate there are 286 row tags in the XML file.   
(Your DataFrame will be a single column, so could also be thought of as a Series.)

In [27]:
import pandas as pd
pd.DataFrame(tag_count).T.sort_values(by=0, ascending=False)


0
row         286
candid      285
candname    285
canclass    285
election    284
officecd    284
primarypay  284
generalpay  284
runoffpay   284
totalpay    284
officedist  245
officeboro   21
response      1

In [ ]:
#Your code here

## Summary

Congratulations! You've started exploring XML data structures used for the web and got to practice data munging and exploring!